pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTLiteralExpression.java

```diff
@@ -87,36 +84,15 @@ public boolean isNull() {
 
     @Override
     public String getImage() {
-        if (isString()) {
-            return ((LiteralExpression.StringVal) node).getValue();
-        } else if (isNull()) {
-            return "";
-        }
-        return node.asCodeString();
+        return expressionToString(node);
     }
 
     public String getName() {
-        /*
-        if (getParent() instanceof ASTNewKeyValueObjectExpression) {
-            ASTNewKeyValueObjectExpression parent = (ASTNewKeyValueObjectExpression) getParent();
-            Optional<NameValueParameter> parameter = parent.node.getParameters().stream().filter(p -> {
-                try {
-                    return this.node.equals(FieldUtils.readDeclaredField(p, "expression", true));
-                } catch (IllegalArgumentException | ReflectiveOperationException e) {
-                    return false;
-                }
-            }).findFirst();
-
-            return parameter.map(p -> {
-                try {
-                    return (Identifier) FieldUtils.readDeclaredField(p, "name", true);
-                } catch (IllegalArgumentException | ReflectiveOperationException e) {
-                    return null;
-                }
-            }).map(Identifier::getValue).orElse(null);
+        if (getParent() instanceof ASTAssignmentExpression && getParent().getParent() instanceof ASTNewKeyValueObjectExpression) {
```

Comment:
Since you reversed engineered this special case, could you put a comment about it?  (Even if this is as simple as  "given Apex code XXX this returns YYY instead of ZZZ".)

Reply:
Could you clarify what you mean by "reverse engineered"? This code is just the Summit-AST equivalent of the previously commented-out code.

Reply:
It's not obvious to me that they are equivalent. Are you asserting that every Summit input must produce the same output as the Jorje original, for each possible execution path? Or can you say something higher-level about what the intention is, that you are preserving?

Reply:
(By "reverse engineered" I was assuming you made a guess about the "intention" and attempted to preserve it. LMK if that wasn't the case.)

Reply:
`ASTNewKeyValueObjectExpression` is a node representing an object constructor with named arguments, e.g.:
```apex
new X(a = 1, b = 2, c = 3)
```
(Only) In this scenario, a literal has the concept of a "name", which this `getName` method is exposing. It seems that Jorje does not build nodes for any of the labels, forcing the previous implementation to use reflection, but my version of the tree builder instead builds the named arguments as an `AssignmentExpression` that contains both the value and the label, making the implementation simpler.

P.S. Although any expression after the label could have a "name", Jorje only has this property for literals.

Reply:
Super: this example and description are exactly what I would hope for in a comment. (Feel free to leave out anything about Jorje if you want to make it forward-looking.)